messagingpush/src/main/java/io/customer/messagingpush/CustomerIOPushReceiver.kt

```diff
@@ -39,49 +46,50 @@ internal class CustomerIOPushReceiver : BroadcastReceiver() {
             context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
         mNotificationManager.cancel(requestCode)
 
-        val bundle = intent.extras
-        val deliveryId = bundle?.getString(DELIVERY_ID_KEY)
-        val deliveryToken = bundle?.getString(DELIVERY_TOKEN_KEY)
+        val payload: CustomerIOParsedPushPayload? = intent.extras?.getParcelable(PUSH_PAYLOAD_KEY)
+        val deliveryId = payload?.cioDeliveryId
+        val deliveryToken = payload?.extras?.getString(DELIVERY_TOKEN_KEY)
```

Comment:
**Adding notes from our discussion on this earlier**

Yes, we can extract delivery the same way. But since we already stored it in `cioDeliveryId`, we decided not to extract again and use one from data class.

Reply:
Yes, updated ✅